shared-libs/search/src/search.js

```diff
@@ -92,16 +99,24 @@ module.exports = function(Promise, DB) {
       skip: 0
     });
 
+    var extendedResults = GenerateSearchRequests.shouldSortByLastVisitedDate(extensions);
     var requests;
     try {
       requests = GenerateSearchRequests.generate(type, filters, extensions);
     } catch (err) {
       return Promise.reject(err);
     }
 
-    return getRows(type, requests, options, extensions)
+    return getRows(type, requests, options, extendedResults)
       .then(function(results) {
-        return _.pluck(results, 'id');
+        if (extendedResults) {
+          return {
+            extendedResults: results.extendedResults,
+            results: _.pluck(results.results, 'id')
+          };
+        } else {
+          return _.pluck(results, 'id');
+        }
```

Comment:
I'm uncomfortable with this returning either an array or an object - it'll confuse me next time I'm in this code. Can we refactor the `else` case to return an object, eg:

```js
        } else {
          return { results: _.pluck(results, 'id') };
        }
```